notifications_utils/clients/statsd/statsd_client.py

```diff
@@ -9,10 +19,32 @@ def __init__(self, host, port, prefix):
         self._port = port
         self._prefix = prefix
         self._sock = socket(AF_INET, SOCK_DGRAM)
+        self._queue = deque()
+
+    def _resolve(self, addr):
+        return gethostbyname(addr)
+
+    @cachetools.func.ttl_cache(maxsize=2, ttl=10, timer=time_monotonic_with_jitter)
+    def _cached_host(self):
+        try:
+            return self._resolve(self._host)
+        except Exception as e:
+            # If we get an error, store `None` in the cache so that we don't keep
+            # trying to retrieve the DNS if DNS server is having issues
+            current_app.logger.warning('Error resolving statsd DNS: {}'.format(str(e)))
+            return None
 
     def _send(self, data):
+        self._queue.append(data)
```

Comment:
Always save data to a queue. This will be the only operation if DNS isn't available right now, and host is cached with the `None` value